pkg/interfaces/contracts/vault/IBasePool.sol

```diff
@@ -45,6 +45,9 @@ interface IBasePool is IPoolSwapStructs {
      * Contracts implementing this function should check that the caller is indeed the Vault before performing any
      * state-changing operations, such as minting pool shares.
      */
+     // enum for transaction type
+    enum OperationType { JOIN, EXIT, SWAP }
```

Comment:
@markusbkoch do we want this feature to apply to weighted pools, or to every pool type?

Reply:
every pool type